lib/modules/showImages.js

```diff
@@ -472,11 +472,15 @@ addModule('showImages', (module, moduleID) => {
 	}
 
 	function updateRevealedImages(type) {
-		for (const image of imageList) {
-			if ($(image).hasClass(type) || image.imageLink.expandOnViewAll) {
-				revealImage(image, findImageFilter(image.imageLink));
-			}
+		let images = imageList.filter(v => $(v).hasClass(type) || v.imageLink.expandOnViewAll);
+
+		if (RESUtils.pageType() === 'linklist') {
+			images = images.map(v => [$(v).closest('.thing').data('rank'), v])
```

Comment:
It would be nice to move `$(v).closest('.thing).data('rank')` to `RESutils.thing::rank` to abstract this for future re-use.